tests/TypeReconciliation/EmptyTest.php

```diff
@@ -417,6 +417,83 @@ function foo(array $arr): void {
                         if (empty($arr["a"])) {}
                     }'
             ],
+            'strlenWithGreaterZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) > 0 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithGreaterZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 0 < strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithGreaterEqualsOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) >= 1 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithGreaterEqualsOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 1 <= strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithInequalZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) !== 0 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithInequalZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 0 !== strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithEqualOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) === 1 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithEqualOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 1 === strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenOnRightHandSide' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 0 < strlen($str) ? $str : "string";
+                    }'
+            ],
```

Comment:
Isn't it identical to `strlenRighthandWithGreaterZero`?

Reply:
In fact, it is. I remove that case.